common/ringpop/ringpop.go

```diff
@@ -116,7 +119,19 @@ func (factory *ringpopFactory) GetMembershipMonitor() (membership.Monitor, error
 func (factory *ringpopFactory) getMembership() (membership.Monitor, error) {
 	var err error
 	factory.monOnce.Do(func() {
-		if rp, err := ringpop.New("temporal", ringpop.Channel(factory.getTChannel()), ringpop.AddressResolverFunc(factory.broadcastAddressResolver)); err != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), persistenceOperationTimeout)
+		defer cancel()
+		ctx = headers.SetCallerInfo(ctx, headers.SystemBackgroundCallerInfo)
```

Comment:
Is SystemBackgroundCallerInfo high priority? We probably should use a high priority one for this call.

Reply:
Hmm, maybe it does not matter because this call is always made during init at which point, there is no other calls to compete with.

Reply:
Yes. I checked the init cluster metadata manager call. It is also sys background. I think this should be fine during startup.